mpc-net/src/multi.rs

```diff
@@ -1,321 +1,372 @@
-use lazy_static::lazy_static;
-use log::debug;
-use rayon::prelude::*;
-use std::fs::File;
-use std::io::{BufRead, BufReader, Read, Write};
-use std::net::{SocketAddr, TcpListener, TcpStream};
-use std::sync::Mutex;
+use std::collections::HashMap;
+use std::error::Error;
+use std::future::Future;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::{TcpListener, TcpStream};
 
 use ark_std::{end_timer, start_timer};
+use async_trait::async_trait;
+use futures::stream::{FuturesOrdered, FuturesUnordered};
+use futures::StreamExt;
+use parking_lot::Mutex;
 
 use super::{MpcNet, Stats};
 
-lazy_static! {
-    static ref CONNECTIONS: Mutex<Connections> =
-        Mutex::new(Connections::default());
-}
-
-/// Macro for locking the FieldChannel singleton in the current scope.
-macro_rules! get_ch {
-    () => {
-        CONNECTIONS.lock().expect("Poisoned FieldChannel")
-    };
-}
-
 #[derive(Debug)]
 struct Peer {
```

Comment:
Yes, that will come with our prodnet implementation.

Since we will be operating on a star network, we can run mutual TLS between the king and every node adjacent to it. By doing this, only nodes that have a valid identity will be able to connect to a network, thus preventing outside nodes from joining. Our party_id will be our index in the deterministically sorted set of peer IDs.